service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java

```diff
@@ -0,0 +1,5 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure;
+
+import bio.terra.workspace.service.resource.controlled.flight.delete.DeleteControlledResourceStep;
+
+public abstract class DeleteAzureControlledResourceStep implements DeleteControlledResourceStep {}
```

Comment:
A note explaining this class' reason for being would be good.

Also, will there be downstream type checking in the azure deletion code to ensure all steps are tagged w/this interface?

Reply:
This class will become much more obvious with a future PR adding behavior here.
I just didn't want to have to change it twice for all the azure resources.

Typechecking for `DeleteControlledResourceStep` is included in this PR. Going further to require `DeleteAzureControlledResourceStep` doesn't really work without also changing the Azure `ControlledResource`s to extend a common class, and I'm not sure it's worth it yet. I was going to start with adding more description here and and also some specifics in `DeleteControlledResourceStep`, that will also direct to here.